chia/data_layer/s3_plugin_service.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import annotations
+
+import asyncio
+import concurrent.futures
+import functools
+import logging
+import sys
+from pathlib import Path
+from typing import Any, Dict, List
+from urllib.parse import urlparse
+
+import boto3 as boto3
+import yaml
+from aiohttp import web
+from botocore.exceptions import ClientError
+
+from chia.types.blockchain_format.sized_bytes import bytes32
+
+log = logging.getLogger(__name__)
+
+
+class S3Plugin:
+    boto_client: boto3.client
+    port: int
+    region: str
+    aws_access_key_id: str
+    aws_secret_access_key: str
+    store_ids: List[str]
+    bukets: Dict[str, List[str]]
+    urls: List[str]
+    instance_name: str
+
+    def __init__(
+        self,
+        region: str,
+        aws_access_key_id: str,
+        aws_secret_access_key: str,
+        store_ids: List[str],
+        bukets: Dict[str, List[str]],
+        urls: List[str],
+        instance_name: str,
+    ):
+        self.boto_client = boto3.client(
+            "s3",
+            region_name=region,
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+        )
+        self.store_ids = store_ids
+        self.bukets = bukets
+        self.urls = urls
+        self.instance_name = instance_name
+
+    async def check_store_id(self, request: web.Request) -> web.Response:
+        self.update_instance_from_config()
+        try:
+            data = await request.json()
+        except Exception as e:
+            print(f"failed parsing request {request} {e}")
+            return web.json_response({"handles_url": False})
+        store_id = bytes32.from_hexstr(data["id"])
+        if store_id.hex() in self.store_ids:
+            return web.json_response({"handles_store": True})
+        return web.json_response({"handles_store": False})
+
+    async def upload(self, request: web.Request) -> web.Response:
+        try:
+            data = await request.json()
+            store_id = bytes32.from_hexstr(data["id"])
+            bucket = self.get_bucket(store_id)
+            try:
+                full_tree_path = Path(data["full_tree_path"])
+                diff_path = Path(data["diff_path"])
```

Comment:
These aren't intended to be part of the exception handling at this level.

```suggestion
            full_tree_path = Path(data["full_tree_path"])
            diff_path = Path(data["diff_path"])
            try:
```

Reply:
what if parsing fails ?

Reply:
With my suggestion it would be categorized as `failed handling request` instead of `failed uploading file to aws` as it is now.  That seems more representative to me.